### ISSUE TITLE
linter:  import/no-cycle error stays in the other file after the cycle is removed (LSP)

### ISSUE THREAD

**Issue by Tholkappiar** [oxc-project/oxc#26809]
### What version of Oxlint are you using?

oxlint 1.83.0, VS Code extension (oxc.oxc-vscode) 1.61.0

### What command did you run?

 VS Code extension (language server), default settings (oxc.lint.run: "onType")

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```json
{
  "plugins": ["import"],
  "categories": { "correctness": "off" },
  "rules": {
    "import/no-cycle": "error"
  }
}
```

### What happened?

**Steps**

1. Create two files that import each other:

   `a.js`
   ```js
   import { b } from "./b.js";
   export const a = 1;
   ```

   `b.js`
   ```js
   import { a } from "./a.js";
   export const b = 1;
   ```

2. Open both files in VS Code. Both show the `import/no-cycle` error. ✅
3. In `b.js`, remove (or comment out) the import and save.

**Expected**

The error goes away in both files, because the cycle is gone.

**Actual**

- `b.js`: error is gone ✅
- `a.js`: error is still shown ❌ (stale)

It only clears after I edit `a.js`, close and reopen it, or run "Oxc: Restart oxlint Server".
Running `oxlint` from the CLI at the same time correctly reports no errors.

**Why (I think)**

The server only re-lints the file that changed. But `import/no-cycle` is a multi-file rule, so a change in one file can change the result in another open file

**Idea**

When a multi-file rule like `import/no-cycle` is enabled, re-validate all open files (not just the changed one) on save or change
